### PR TITLE
clusterID is not set when build cluster key

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/cluster_builder.go [hunk 1]
```diff
@@ -318,24 +318,34 @@ func (cb *ClusterBuilder) buildDefaultCluster(name string, discoveryType cluster
...
+	//
+	// http2 identifies if the cluster is for an http2 service
 	http2          bool
 	downstreamAuto bool
+
+	// Dependent configs
+	//
+	service         *model.Service
+	destinationRule *config.Config
+
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster_builder.go:343]: This is to sort them according to the category

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 2]
```diff
@@ -138,14 +138,15 @@ func buildClusterKey(service *model.Service, port *model.Port, cb *ClusterBuilde
 	clusterName := model.BuildSubsetKey(model.TrafficDirectionOutbound, "", service.Hostname, port.Port)
 	clusterKey := &clusterCache{
 		clusterName:     clusterName,
-		service:         service,
-		networkView:     cb.proxy.GetNetworkView(),
-		pushVersion:     cb.req.Push.PushVersion,
-		destinationRule: cb.req.Push.DestinationRule(cb.proxy, service),
 		locality:        cb.proxy.Locality,
+		proxyClusterID:  string(cb.proxy.Metadata.ClusterID),
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster.go:142]: added this

**Review by ramaraochavali — APPROVED** [review]

File: pilot/pkg/networking/core/v1alpha3/cluster_builder.go [hunk 3]
```diff
@@ -318,24 +318,34 @@ func (cb *ClusterBuilder) buildDefaultCluster(name string, discoveryType cluster
...
 	// Push version is a very broad key. Any config key will invalidate it. Its still valuable to cache,
 	// as that means we can generate a cluster once and send it to all proxies, rather than N times for N proxies.
 	// Hypothetically we could get smarter and determine the exact set of all configs we use and their versions,
 	// which we probably will need for proper delta XDS, but for now this is sufficient.
 	pushVersion string
-	service     *model.Service
-	networkView map[network.ID]bool
 }
 
 func (t *clusterCache) Key() string {
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/cluster_builder.go:351]: I think you need to add it to here to do anything?

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/cluster_builder.go:351]: nevermind, already there

**Pull request merged**